module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -737,16 +736,44 @@ protected function normalizeUrlForComparison($url)
         return trim(end($parts), '/');
     }
 
+    /**
+     * Checks if a followup url is set
+     *
+     * @return bool
+     */
+    protected function hasFollowupUrl()
+    {
+        return null !== $this->followup()->retrieve('url');
+    }
+
     /**
      * Retrieve a referer to keep post-login redirect pointing
      * to an appropriate location.
      * Unset the followup before returning.
      *
+     * @param bool $checkRedirect Whether the query should be checked for param 'redirect'
+     *
      * @return string
      */
-    protected function getFollowupUrl()
-    {
-        return $this->followup()->retrieve('url', '');
+    protected function getAndClearFollowupUrl($checkRedirect = false)
+    {
+        if ($url = $this->followup()->retrieveAndClear('url', '')) {
+            $lightboxParent = $this->followup()->retrieveAndClear('lightboxParent');
+            // If a user clicks on the "Your Account" link, we want to be sure
+            // they get to their account rather than being redirected to an old
+            // followup URL. We'll use a redirect=0 GET flag to indicate this:
+            if (!$checkRedirect || $this->params()->fromQuery('redirect', true)) {
+                if (null !== $lightboxParent && $this->getAuthManager()->getSessionInitiator($this->getServerUrl())) {
```

Comment:
Does the authentication method actually matter here? Or should we instead be checking whether the request is already in a lightbox or not? In practice, this scenario currently only arises from SSO, but that may be something of a coincidence, and the thing that really impacts the desired behavior is whether we're already in a lightbox or need to restore a lightbox from prior to a redirect. Sorry if I'm off base here, but I think there might be an opportunity to generalize the code further.

Reply:
You are absolutely right. Thanks for the hint!  I changed this.

Reply:
Great, thanks -- I'll give this one more thorough test and review tomorrow to see if I can think of or find any other issues, but I have a feeling we may be done. More soon!